src/components/UserContext/index.tsx

```diff
@@ -0,0 +1,30 @@
+'use client';
+
+import { generateContext } from '@/react-utils';
+import { useState } from 'react';
+
+interface UserContextProps {
+  loginId: number | null;
+  setLoginId: (id: number | null) => void;
+}
+
+const [UserProvider, useUserContext] = generateContext<UserContextProps>({
+  name: 'user-info',
+});
+
+export function UserProviderWrapper({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [loginId, setLoginId] = useState<number | null>(null);
+
+  const contextValue = {
+    loginId,
+    setLoginId,
+  };
```

Comment:
`useMemo`를 사용하셔야 해요.

```suggestion
  const contextValue = useMemo(() => ({
    loginId,
    setLoginId,
  }), [loginId, setLoginId]);
```